exporters/dsexporter/dsexporter.go

```diff
@@ -3,37 +3,69 @@ package main
 import (
 	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	appstudioProbeSuccess *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		appstudioProbeSuccess: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "appstudio_probe_success",
+			Help: "Availability of appstudio service/component",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.appstudioProbeSuccess.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	e.appstudioProbeSuccess.Collect(ch)
+}
+
+func availabilityHandler(response http.ResponseWriter, req *http.Request, exporter *CustomCollector) {
+	service := req.URL.Query().Get("service")
+	check := req.URL.Query().Get("check")
+	availability := checkIfServiceUp(service, check)
+
+	exporter.appstudioProbeSuccess.WithLabelValues("service", "check").Set(availability)
+	fmt.Fprintf(response, "Service: %s, Check Name: %s, Availability: %f", service, check, availability)
+}
+
+func checkIfServiceUp(service, check string) float64 {
+	serviceUrl := "http://" + service + ".com"
```

Comment:
Building the url like this makes a lot of assumptions. It'd be simpler to pass the url in directly.

Also, please use `https` by default.